### ISSUE TITLE
Secondary alpha value for conditionals (i.e. "fade friendly NPCs" > fade to secondary value)

### ISSUE THREAD

**Issue by kesava-wow** [kesava-wow/kuinameplates2#75]
Or maybe "fade non-targets to..." would work. Just fade everything which isn't the target to that, fade things which match the conditions to faded alpha. "Fade by default" uses non-target value.

**Comment by benoitheylens** [comment]
Much needed and appreciated feature :).
Thank you !

**Comment by benoitheylens** [comment]
"Bump". I hate to have my screen cluttered with non-wanted plates, so I want to see only Quest givers and helpfull NPC when I am not in combat (with your sweet name-only plates) and when I enter combat, see all enemy plates (and even friendly ones if the addon requires it).

I tried with the "when in combat : show" but that's not working.

I mean, it works when I play with "namesonly" mode, not with plates. I think I am not the only one here to want this feature.

Regards.

**Comment by benoitheylens** [comment]
UP, again. I got massive performance loss in Dalaran and in Suramar (good rig with 970 GTX and 16GB ram, ssd) when there is too manyplates on screen. I really would like to be able to hide non-wanted names ...

**Comment by kesava-wow** [comment]
This wouldn't have any effect on FPS.

**Comment by benoitheylens** [comment]
It would a least for visual comfort. Am I the only one to find any utility to not clutter my screen with useless information ? Is this a Blizzard thing ? Do I have to get rid off "show every frame" option ? I am feeling like I need a PhD in plate-addons to get what I want (It worked so good back in WoD, I don't understand what changed except the new Blizzard option).

I am sorry if I pollute this thread with my hurted feelings, but I really want this to work, your addon is everything WoW should have in the first place ...

**Comment by kesava-wow** [comment]
Arbitrary nameplates can't actually be hidden (in a way which would increase performance) besides the checkboxes under ESC > interface > names and the hotkeys. So yes, it's a blizzard thing. Making them invisible (which this would do as a workaround for that) means the click box still exists - so you might sometimes click a nameplate you can't see when you're trying to click something you can see - and that the performance impact from having the nameplate render still exists.

What do you actually want again, hidden friendly NPCs? You couldn't do that in WOD either.

**Comment by benoitheylens** [comment]
Back in WoD, i only activated plates on enemies (v). But then, i only showed names (not plates) on friendly NPC that I wanted to follow (quest givers or interractive).

Now, with your "name-only" mode, I understand you just replace the standard name by a "nameplate" that is moded to look like it's only a name. And to match the previous blizz option, I set the "fade" NPC to 0. So my screen isn't cluttered with unwanted names.

The problem is, when I enter in combat, then all nameplates but the target fades to 0. That's a real problem. 

The "workaround" I would do is to switch the fade value to 1 when I enter combat mode, and back to 0 when I leave combat. 

The performance would not be affected, as I understand because the nameplate is rendered then alpha'ed to 0 (but still there), but at least, I would get a clean view of what matters to the screen. 

And I am surprised that people do not think like I do, actually :D.

Tldr; 

I need the names of players and interractive players, with your cute way to render it.
I need all enemies frames at all time, to get a sense of what danger is around me (be it a giant spider or an ugly orc :p ), even when I am in combat (how to check if my dots, CC, etc are on other targets than my main target, otherwise) ?

Sorry if it seems to you like I am a pain to deal with, but I really love your addon and praised it to my guildmates (my tank did build a statue in your name) and I want it to work to meet our needs :). 

Regards and - again - thank you for all your work.

**Comment by kesava-wow** [comment]
Name only mode doesn't replace the floating names, its just an alternate way of drawing nameplates. The floating names can't be modified by addons. Thats why you'll still sometimes see them for unattackable enemies. 

You should be able to replicate the setup you had in WOD by disabling friendly nameplates as you did before (except now you have to use the hotkey since for some reason they removed the checkbox for it). Then you can still only show floating names on quest npcs and such with the "npc names" drop-down under esc > interface > names.

I understand the problem of everything fading out to 0 when you have a target, but the capability to do what you were doing in wod hasn't been removed. If you dont want to do that anymore - I.e. you want my nameplates (in name only mode) for friendlies - then yes, you'll need to wait for me to do this.

Thanks.

**Comment by benoitheylens** [comment]
yes I understand that fully :).
I Need your cute "only-name-frames" for friendly and nameplates with healthbars for enemies (that, I can achieve now), but I need the frames to stay visible when in combat ...

I'll guess I'll have to wait. 

How many beers to get you on track :D ?

**Comment by ajhix36** [comment]
This works perfectly now with the new option in 14-2, to avoid fading tracked or in-combat nameplates.
